Microsoft.Toolkit.Uwp.UI.Animations/Xaml/Interfaces/IInternalImplicitAnimation.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+#nullable enable
+
+namespace Microsoft.Toolkit.Uwp.UI.Animations
+{
+    /// <summary>
+    /// An interface adding notification support to implicit animations. This is needed to
+    /// avoid the type parameters when the event is subscribed to from <see cref="ImplicitAnimationSet"/>.
+    /// </summary>
+    internal interface IInternalImplicitAnimation
```

Comment:
Do we need this interface? Its internal and only one implementer. Why do we want to avoid the type parameters?

Reply:
It's not that we want to avoid type parameters, it's that we'd have no way of knowing which ones they'd be at runtime from there. Imagine we had some `ImplicitAnimationSet`, which registers this handler when it's created:

https://github.com/windows-toolkit/WindowsCommunityToolkit/blob/85005e660cd4456f67d3d90243d958f43ea9b19f/Microsoft.Toolkit.Uwp.UI.Animations/Xaml/ImplicitAnimationSet.cs#L60-L65

Now, the handler has absolutely no way of knowing the actual runtime type of each animation that's added or removed, yet it needs to access some instance members of a generic class, without knowing what the type arguments are. The current animation could be an `ImplicitAnimation<double?, double>`, an `ImplicitAnimation<string, Vector3>`, or something else. By using the interface we can effectively just strip out the generic parameters entirely and just access the event directly (with no reflection). So it's not really that we "need" an interface per se, it's more just a trick to be able to get a non-generic way to access an instance member in a generic class, without knowing the type arguments.

For reference, it's the same kind of trick I'm using in `StrongReferenceMessenger` in the MVVM Toolkit to be able to completely avoid reflection and yet access types from generic classes, from a non-generic context.

Let me know if this clears things up! 🙂